internal/policy/container/trademark_validator.go

```diff
@@ -0,0 +1,20 @@
+package container
+
+import (
+	"regexp"
+	"strings"
+)
+
+// ViolatesRedHatTrademark validates if a string meets specific "Red Hat" naming criteria
+func ViolatesRedHatTrademark(nameLabel string) bool {
+	// string starts with Red Hat variant
+	startingWithRedHat := regexp.MustCompile("^[^a-z0-9]*red[^a-z0-9]*hat").MatchString(strings.ToLower(nameLabel))
+
+	// string contain Red Hat variant (not starting with)
+	containsRedHat := regexp.MustCompile("red[^a-z0-9]*hat").MatchString(strings.ToLower(nameLabel))
+
+	// string contains "for Red Hat" variant
+	containsForRedHat := regexp.MustCompile("for[^a-z0-9]*red[^a-z0-9]*hat").MatchString(strings.ToLower(nameLabel))
+
+	return startingWithRedHat && containsRedHat && !containsForRedHat
```

Comment:
Should `containsRedHat && !containsForRedHat)` be grouped? I'm understanding `containsForRedHat` to mean that you *may* use "for Red Hat" in your naming, but if you don't, doesn't MatchString return "false", making the return condition return "true" for containsRedHat?

Reply:
I think you mean (containsRedHat && !containsForRedHat) correct? This is what I initially had but I wasn't sure.

Reply:
Yeah, I mean - I also think that might be incomplete.

I'd guess it's possible for a name to be something like "ProductName By Red Hat for Red Hat". Technically, still a violation, and I think the existing logic would not catch it, even with the grouping.

I'm guessing you want a count of matches, and a count of "for Red Hats". Subtract the two, that's the number of violations for the last two cases.

Reply:
I updated this.